tools/Customization/DevHome.Customization/ViewModels/DevDriveInsights/OptimizeDevDriveDialogViewModel.cs

```diff
@@ -46,6 +46,15 @@ public partial class OptimizeDevDriveDialogViewModel : ObservableObject
     [ObservableProperty]
     private string _directoryPathTextBox;
 
+    [ObservableProperty]
+    private bool _isPrimaryButtonEnabled;

```

Comment:
You don't need IsPrimaryButtonEnabled AND IsNotDevDrive, since they're always the inverse of each other. I suggest having one `IsDevDriveDirectorySelected` property. Bind ContentDialog.IsPrimaryButtonEnabled to it, and then bind the grid visibility to it using a BoolNegationConverter.

Reply:
There is an issue with using 1 bool. The error message will show up by default when the user has not entered anything in the text box.

Reply:
Can you bind it to ErrorMessage not being null?

Reply:
Then the primary button would be enabled if the input textbox is empty. Don't think that is ok.

Reply:
I see. Yep, this seems reasonable then 🙂